test/test_ccf.py

```diff
@@ -172,6 +190,32 @@ def test_submit_claim_notary_x509(
     client.submit_claim(claim_with_receipt)
 
 
+def test_untrusted_ca(client: Client):
+    """
+    Submit a claim signed by a certificate issued by an untrusted CA.
+    """
+    untrusted_ca = X5ChainCertificateAuthority(kty="ec")
+    identity = untrusted_ca.create_identity(length=1, alg="ES256", kty="ec")
+    claims = crypto.sign_json_claimset(identity, {"foo": "bar"})
+
+    with pytest.raises(ServiceError, match="Certificate chain is invalid"):
+        client.submit_claim(claims)
+
+
+def test_root_identity(
+    client: Client,
+    trust_store: TrustStore,
+    trusted_ca: X5ChainCertificateAuthority,
+):
+    """
+    Submit a claim signed by the self-signed root certificate.
```

Comment:
A certificate with the CA bit set should never be used for anything other than signing certificates that are issued. The fact that we allow this is a bug. In the trust store, there should only be CA certificates or self-signed non-CA certificates.